torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -90,21 +108,25 @@ static Variable applySlice(const Variable& self, int64_t dim, PyObject* slice, b
   return self.slice(dim, start, stop, step);
 }
 
-static Variable applySelect(const Variable& self, int64_t dim, int64_t index, int64_t real_dim=0) {
-  if (index == 0 && dim == 0 && self.dim() == 0) {
+static Variable applySelect(const Variable& self, int64_t dim, PyObject* index, int64_t real_dim=0) {
+  auto& var = THPVariable_Unpack(index);
```

Comment:
I don't think `index` is necessarily a Variable (Tensor) here. It might be a Python number for example.

Reply:
Yes, updated this